lib/streamlit/web/server/authlib_tornado_integration.py

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from authlib.integrations.base_client import FrameworkIntegration
+
+
+class TornadoIntegration(FrameworkIntegration):
+    def update_token(self, token, refresh_token=None, access_token=None):
+        pass
+
+    @staticmethod
+    def load_config(oauth, name, params):
+        if not oauth.config:
+            return {}
+
+        rv = {}
+        for k in params:
+            v = oauth.config.get(name, {}).get(k, None)
+            if v is not None:
+                rv[k] = v
+        return rv
```

Comment:
What do these things do? I assume it's authlib specific. Will benefit from some comments.

Reply:
Yep, I refactored it a little bit to have better names and docsting.
In a nutshell, we need to override how authlib reads configs to be able to specify multiple auth providers based on secrets.toml definition